apis/r/R/SOMAArrayBase.R

```diff
@@ -34,7 +34,11 @@ SOMAArrayBase <- R6::R6Class(
           NULL
       } else {
           rl <- sr_next(private$soma_reader_pointer)
-          private$soma_reader_transform(rl)
+          if (is.null(rl[[1]])) {
+            rl
```

Comment:
Returning `rl` is probably not ideal.

Reply:
And before `rl <- sr_next(....)` consider adding a check on `sr_complete(private$soma_reader_pointer)`

Reply:
@eddelbuettel if at end of stream, would `return(NULL)` be preferable?

Reply:
It all depends. I opted to return a list with two NULLs as the normal object is a list with two pointers (passed to Arrow).

The best idiom really is to to first check with `sr_complete()` and then quietly break.  So yep, `return(invisible(NULL))` is a good choice.